lib/fake_ifdh.py

```diff
@@ -94,21 +102,19 @@ def checkToken(tokenfile: str) -> bool:
     """check if token is (almost) expired"""
     if not os.path.exists(tokenfile):
         return False
-    exp_time = None
-    cmd = f"decode_token.sh -e exp {tokenfile} 2>/dev/null"
-    with os.popen(cmd, "r") as f:
-        exp_time = f.read()
     try:
-        add_event(f"expiration: {exp_time}")
-        return int(exp_time) - time.time() > 60
-    except ValueError as e:
-        print(
-            "decode_token.sh could not successfully extract the "
-            f"expiration time from token file {tokenfile}. Please open "
-            "a ticket to Distributed Computing Support if you need further "
-            "assistance."
-        )
-        raise
+        exp_time = None
+        with open(tokenfile) as f:
+            token_encoded = f.read(4096).strip()
+            token = dict(
+                scitokens.SciToken.deserialize(token_encoded, insecure=True).claims()
+            )
+            exp_time = token["exp"]
+            add_event(f"expiration: {exp_time}")
+            return int(exp_time) - time.time() > 60
+    except Exception:
+        # anything else, just say its bad
+        return False
```

Comment:
Is this the right place to catch the exception? Seems the caller is in a better position to handle it. Like in `getToken()`, with the first call we don't care why it's invalid since we'll get a new one, but with the second call we should pass along why the newly-acquired token is not valid.

Reply:
We want it to just return False; the scitokens library (and underlying JWT library0 raises different exceptions for expired, bad signature, empty, etc...  Maybe we want a flag passed in that says if we should return false or re-raise?  We could use the former behavior when deciding if we should call htgettoken, and the latter to check the token the second time, so it raises an exception about what was wrong?

Reply:
>  Maybe we want a flag passed in that says if we should return false or re-raise? 

I like this idea.